### PR TITLE
Feature/logger

### PR DESCRIPTION
**Pull request by gabrielPeart**
- Logger
- Code clean up
- Fixed failing tests
- Fixed lint issues

### CODE REVIEW

File: Sources/ShortcutFoundation/Core/Logger/Loggable.swift [hunk 1]
```diff
@@ -0,0 +1,145 @@
+//
+// Loggable.swift
+// VSFoundation
+//
+// Created by Gabriel Sabadin on 2021-12-13.
+// Copyright Virtual Stores - 2021
```
Comment (lemonandlime) [inline Sources/ShortcutFoundation/Core/Logger/Loggable.swift:6]: These dam headers sometimes I feel that our standard should just be to not have them at all.

File: Tests/ShortcutFoundationTests/StringExtensionTests.swift [hunk 2]
```diff
@@ -11,12 +11,11 @@ final class StringExtensionTests: XCTestCase {
     }
     
     func test_negative_futute_age_from_personal_number() {
-
         let dateOfBirth = Calendar.current.date(from: .init(year: 1985, month: 12, day: 23))!
         let expectedAge = Calendar.current.dateComponents([.year], from: dateOfBirth, to: .now).year
         let sut = "198512230000"
         let age = sut.ageFromPersonalNumber
-
+        // in December 2022
```
Comment (lemonandlime) [inline Tests/ShortcutFoundationTests/StringExtensionTests.swift:18]: This is now dynamic so no special date here.

**Review by lemonandlime — APPROVED** [review]

**Comment by lemonandlime** [comment]
Nice! Thx for the logger.

File: Sources/ShortcutFoundation/Core/Logger/Loggable.swift (hunk 1 above)
Comment (gabrielPeart) [reply Sources/ShortcutFoundation/Core/Logger/Loggable.swift]: My bad, I was sloppy here. Thanks for checking it out

**Comment by gabrielPeart** [comment]
> Nice! Thx for the logger.

I have solved the issues

**Comment by gabrielPeart** [comment]
@darya-gurinovich I have added part of your logic, but I abstracted it to be compatible with older iOS versions as well

File: Sources/ShortcutFoundation/Core/Logger/Loggable.swift [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+            strategy = AppleLogger(verbosity: verbosity, identifier: identifier, category: category)
+        } else {
+            strategy = PrintLogger(verbosity: verbosity, identifier: identifier, category: category)
+        }
+    }
+
+    private func makeStrategy(identifier: String = "",
+                              category: String = "") -> Loggable {
+        return PrintLogger(verbosity: verbosity, identifier: identifier, category: category)
+    }
```
Comment (darya-gurinovich) [inline Sources/ShortcutFoundation/Core/Logger/Loggable.swift:140]: Where can we use this method?

**Comment by gabrielPeart** [comment]
@lemonandlime @darya-gurinovich @AndreasShortcut @sheikhbayazid 

to use the logger, I recommend the following:

1. In your IEnvironmentConfig you can define the verbosity accordingly.
2. Inject your logger in your AppConfig
3. When you want to use the logger just go for: `@LazyInject private var logger: Loggable`

**Review by darya-gurinovich — COMMENTED** [review]
Great code! Thanks for using some of my stuff!

File: Sources/ShortcutFoundation/Core/Logger/Loggable.swift (hunk 3 above)
Comment (gabrielPeart) [reply Sources/ShortcutFoundation/Core/Logger/Loggable.swift]: @darya-gurinovich nowhere, I was considering removing the IF statement to this method, but changed my mind and forgot to delete it. Thanks for noticing.

**Comment by darya-gurinovich** [comment]
> @lemonandlime @darya-gurinovich @AndreasShortcut @sheikhbayazid
> 
> to use the logger, I recommend the following:
> 
> 1. In your IEnvironmentConfig you can define the verbosity accordingly.
> 2. Inject your logger in your AppConfig
> 3. When you want to use the logger just go for: `@LazyInject private var logger: Loggable`

Thanks! What should we do if we want to use different levels of the verbosity?

**Comment by gabrielPeart** [comment]
> > @lemonandlime @darya-gurinovich @AndreasShortcut @sheikhbayazid
> > to use the logger, I recommend the following:
> > 
> > 1. In your IEnvironmentConfig you can define the verbosity accordingly.
> > 2. Inject your logger in your AppConfig
> > 3. When you want to use the logger just go for: `@LazyInject private var logger: Loggable`
> 
> Thanks! What should we do if we want to use different levels of the verbosity?

I have improved the API: https://github.com/shortcut/shortcut-foundation-ios/pull/21/commits/f9fac01c5d0a5ebbdbf2259323527ffae6496e0e

```Swift
// utilizing the global defined verbosity
logger.log(message: "some message")
        
// utilizing a defined verbosity for this particular log
logger.log(message: "some critical message", verbosity: .critical)
```

**Comment by lemonandlime** [comment]
We could create a global function just like print() so it's very compact. 

Also I think we could play around with build schemes to set the log level verbosity 

Great this is a really nice feature

**Comment by gabrielPeart** [comment]
> We could create a global function just like print() so it's very compact. 
> 
> Also I think we could play around with build schemes to set the log level verbosity 
> 
> Great this is a really nice feature 

Yes, we can also create a global function. Feel free to do it if you want to.

**Pull request merged**